core/src/main/java/org/eclipse/hono/util/TenantObject.java

```diff
@@ -55,7 +59,14 @@
     @JsonIgnore
     private Map<String, JsonObject> adapterConfigurations;
     @JsonIgnore
-    private TrustAnchor trustAnchor;
+    private List<JsonObject> trustedCaList;
+    @JsonIgnore
+    private List<TrustAnchor> trustAnchors;
```

Comment:
FMPOV this is the only collection we need. What are the others for?

Reply:
a client might be interested to get all the trusted CAs in a tenant's trust-store. the `trustedCaList` is a cache for subsequent calls to `getTrustConfigurations()`

likewise, the `subjectNames` is a cached for subsequent calls to `getTrustedCaSubjectNames()`

i find them worthwhile to have. Don't you agree?

Reply:
FMPOV we should keep the use cases in mind that the `TenantObject` is relevant for:
1. create/update a tenant in the device registry and
2. verify a client certificate in a protocol adapter

So, for use case 1, the registry implementation needs be check if any of the subject DNs are already in use by another tenant. So, having a *getTrustedCaSubjectDns* method returning a list of Strings should do. However, this is done only once when the request to create/update is received, so I do not quite see why we should cache this list instead of just iterating over the *trusted-ca* property content and retrieve the subject DNs ...

For use case 2, the protocol adapter needs to
1. look up the tenant by subject DN, getting back the tenant data,
2. retrieve the list of TrustAnchors from the tenant data and
3. pass the list of trust anchors to the CertificateValidator which uses it to veryfy the client certificate.

Looks like for that we only need the *trustAnchors*, don't we?
Based on this, maybe you can re-assess all the methods and fields that you have added?

Reply:
thanks for this elaborate feedback 👍

> Based on this, maybe you can re-assess all the methods and fields that you have added?

Sure